From fa720f79cb80b51a0798144b6f310ed1e1938478 Mon Sep 17 00:00:00 2001
From: Tim Gardner <tim.gardner@canonical.com>
Date: Tue, 13 Jul 2010 06:41:16 -0600
Subject: [PATCH] UBUNTU: SAUCE: fix build error with CONFIG_BLK_DEV_INITRD=n

The original aynchronous boot patch (UBUNTU: SAUCE: Make populate_rootfs asynchronous)
did not take into consideration the case when CONFIG_BLK_DEV_INITRD=n,
e.g., populate_rootfs_domain becomes undefined. Therefore, add it to
noinitramfs.c where its use is benign.

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
Original-patch-by: Nicolas Pitre <nicolas.pitre@linaro.org>
Cc: Surbhi Palande <surbhi.palande@canonical.com>
Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>
---
 init/noinitramfs.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/init/noinitramfs.c b/init/noinitramfs.c
index 267739d..e0f648c 100644
--- a/init/noinitramfs.c
+++ b/init/noinitramfs.c
@@ -22,6 +22,8 @@
 #include <linux/kdev_t.h>
 #include <linux/syscalls.h>
 
+LIST_HEAD(populate_rootfs_domain);
+
 /*
  * Create a simple rootfs that is similar to the default initramfs
  */
-- 
1.7.0.4

